tests/providers/snowflake/hooks/test_snowflake.py

```diff
@@ -16,340 +16,243 @@
 # specific language governing permissions and limitations
 # under the License.
 #
-import os
+import pathlib
 import re
 import unittest
+from pathlib import Path
 from unittest import mock
 
+import pytest
 from cryptography.hazmat.backends import default_backend
 from cryptography.hazmat.primitives import serialization
 from cryptography.hazmat.primitives.asymmetric import rsa
-from parameterized import parameterized
+from py._path.local import LocalPath
```

Comment:
Let’s use this oppertunity to also migrate from `tmpdir` to `tmp_path` (which is based on pathlib).

Reply:
Done. https://github.com/apache/airflow/pull/20095/commits/6f164b9016fb4c543958b7a7fb517ed80e26aa23